core/indigo-core/molecule/ket_commons.h

```diff
@@ -98,5 +99,106 @@ namespace indigo
         std::list<KETTextLine> _block;
         Vec3f _pos;
     };
+
+    class KETReactionArrow : public MetaObject
+    {
+    public:
+        static const std::uint32_t cid = "KET reaction arrow"_hash;
+        enum
+        {
+            EOpenAngle,
+            EFilledTriangle,
+            EFilledBow,
+            EDashedOpenAngle,
+            EFailed,
+            EBothEndsFilledTriangle,
+            EEquilibriumFilledHalfBow,
+            EEquilibriumFilledTriangle,
+            EEquilibriumOpenAngle,
+            EUnbalancedEquilibriumFilledHalfBow,
+            EUnbalancedEquilibriumLargeFilledHalfBow,
+            EUnbalancedEquilibriumFilleHalfTriangle
+        };
+
+        KETReactionArrow(int arrow_type, const Vec2f& begin, const Vec2f& end) : MetaObject(cid), _arrow_type(arrow_type), _begin(begin), _end(end){};
+
+        MetaObject* clone() const override
+        {
+            return new KETReactionArrow(_arrow_type, _begin, _end);
+        }
+
+        int _arrow_type;
+        Vec2f _begin;
+        Vec2f _end;
+    };
+
+    class KETReactionPlus : public MetaObject
+    {
+    public:
+        static const std::uint32_t cid = "KET reaction plus"_hash;
```

Comment:
Maybe express that this is a constant by naming it ALL_CAPS: `CID` ?

Reply:
fixed